src/include/simeng/branchpredictors/BranchPrediction.hh

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cstdint>
+
+namespace simeng {
+
+/** The types of branches recognised. */
+enum class BranchType {
+  Conditional = 0,
+  LoopClosing,
+  Return,
+  SubroutineCall,
+  Unconditional,
+  Unknown
+};
+
+/** A branch result prediction for an instruction. */
+struct BranchPrediction {
+  /** Whether the branch will be taken. */
+  bool isTaken;
+
+  /** The branch instruction's target address. If `isTaken == false`, the value
+   * will be ignored. */
+  uint64_t target;
+
+  /** Check for equality of two branch predictions . */
+  bool operator==(const BranchPrediction& other) {
+    if ((isTaken == other.isTaken) && (target == other.target))
+      return true;
+    else
+      return false;
```

Comment:
Function body can be condensed to `return ((isTaken == other.isTaken) && (target == other.target));`